src/main/java/pl/med/clinic/service/VisitServiceImpl.java

```diff
@@ -15,11 +18,38 @@
 public class VisitServiceImpl implements VisitService {
 
     private final VisitRepository visitRepository;
+    private final PatientRepository patientRepository;
 
     @Override
-    public VisitsDtoResponse get(Long patientId) {
+    public VisitsDtoResponse getPatientsVisits(Long patientId) {
         List<VisitEntity> visits = visitRepository.getVisitEntitiesByPatientId(patientId);
         return new VisitsDtoResponse(visits);
     }
 
+    @Override
+    public VisitDto getVisit(Long visitId) {
+        return new VisitDto(visitRepository.getOrThrow(visitId));
+    }
+
+    @Override
+    public VisitDto createVisit(VisitDtoRequest newVisit) {
+        VisitEntity visitEntity = mapToEntity(newVisit);
+        VisitEntity save = visitRepository.save(visitEntity);
+        return new VisitDto(save);
```

Comment:
nie widzę miejsca gdzie dodajesz relację do pacjenta ??
Wywal patientId z konstruktora Visit a relacje dodaj przez setter.
To samo zrób po stronie patientEntity, tzn dodaj visit przez metodę ale nie przez setter (zastanów się dlaczego tak)

Reply:
do tego wróćmy - mam problem z dodaniem relacji przez metodę, powinnam rozszerzyć listę przez add w odpowiednim PatientEntity (tak mi się wydaje), ale nie wiem jak zrobić to na liście będącej elementem obiektu, jeszcze nad tym posiedzę, ale na razie wysyłam bez

Reply:
dodałam dwie metody na dodanie relacji do pacjenta - add do listy i usuniecie racji - remove z listy,
do poprawki - napewno można to napisać lepiej, jeśli wogóle jest dobrze,
do przegadania